Claude/tts web interface 011 c uy zbv5s9c4 s gcru1 u1 kc

Jayzed1691 opened the pull request.

claude committed:
Add Streamlit web interface for TTS with voice cloning

Implemented a comprehensive web interface using Streamlit that provides:

Features:
- Voice cloning for English using XTTS v2 (3+ seconds reference audio)
- Standard Indonesian TTS using Fairseq VITS
- Long-form text synthesis with automatic sentence splitting
- Multiple reference audio file support for improved cloning quality
- GPU acceleration with automatic detection
- Audio preview, playback, and download functionality
- Real-time processing feedback and statistics
- Optional waveform visualization

Files added:
- streamlit_tts_app.py: Main Streamlit application with full UI
- requirements.streamlit.txt: Additional dependencies for web interface
- STREAMLIT_TTS_GUIDE.md: Comprehensive documentation and guide
- STREAMLIT_QUICKSTART.md: Quick start guide for new users

The interface supports both English (with voice cloning) and Indonesian
(standard synthesis), enabling users to clone voices from audio samples
and generate speech from long-form text through an intuitive web UI.

Usage:
  pip install -r requirements.streamlit.txt
  streamlit run streamlit_tts_app.py

claude committed:
Fix NumPy version compatibility issue with Numba

Fixed ImportError: "Numba needs NumPy 2.2 or less. Got NumPy 2.3"

Changes:
- Updated requirements.txt to cap NumPy at <2.0 for Python >3.10
- Updated requirements.streamlit.txt with NumPy version constraint
- Added comprehensive TROUBLESHOOTING.md guide
- Updated STREAMLIT_QUICKSTART.md with installation fix
- Updated STREAMLIT_TTS_GUIDE.md with dependency warning

The issue occurred because Numba (required by librosa, which is used
by TTS) does not support NumPy 2.3+. The requirements now enforce
NumPy <2.0 to prevent this compatibility issue.

Users experiencing this error should run:
  pip install "numpy>=1.24.3,<2.0"

Or reinstall from requirements:
  pip install -r requirements.streamlit.txt

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/coqui-ai/TTS?pullRequest=4362) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/coqui-ai/TTS?pullRequest=4362) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/coqui-ai/TTS?pullRequest=4362) it.</sub>

claude committed:
Clarify installation order in documentation

Updated all documentation to emphasize that the base TTS package
must be installed BEFORE running the Streamlit app.

Changes:
- STREAMLIT_QUICKSTART.md: Added clear 3-step installation process
- STREAMLIT_TTS_GUIDE.md: Enhanced Step 1 with warnings and details
- TROUBLESHOOTING.md: Added "No module named 'coqpit'" error section

The "No module named 'coqpit'" error occurs when users try to run
the Streamlit app without first installing the base TTS package.
All documentation now clearly states the installation order:
  1. pip install -e .[all]  (TTS package)
  2. pip install -r requirements.streamlit.txt  (Streamlit)
  3. streamlit run streamlit_tts_app.py  (Run app)

claude committed:
Fix PyTorch 2.6 compatibility with weights_only parameter

Resolved "Weights only load failed" error in PyTorch 2.6+ by adding
weights_only=False parameter to all torch.load() calls.

Changes:
Core TTS files patched:
- TTS/utils/io.py: Main model loading function (load_fsspec)
- TTS/tts/utils/managers.py: ID manager file loading
- TTS/tts/layers/xtts/xtts_manager.py: Speaker manager loading
- TTS/tts/layers/xtts/hifigan_decoder.py: HiFiGAN checkpoint loading
- TTS/tts/layers/xtts/dvae.py: DVAE mel norms loading
- TTS/tts/layers/xtts/trainer/gpt_trainer.py: GPT and DVAE checkpoints

Documentation updated:
- TROUBLESHOOTING.md: Added PyTorch 2.6 error section with solutions
- STREAMLIT_QUICKSTART.md: Added PyTorch compatibility warning
- STREAMLIT_TTS_GUIDE.md: Added PyTorch version notes

Background:
PyTorch 2.6 changed the default value of weights_only in torch.load
from False to True for security. Coqui TTS model checkpoints contain
custom classes (like XttsConfig) that require weights_only=False to
unpickle properly.

All torch.load calls now explicitly set weights_only=False to maintain
compatibility with PyTorch 2.6+ while preserving the ability to load
TTS model checkpoints with custom classes.

claude committed:
Fix transformers 4.50+ compatibility by inheriting from GenerationMixin

Resolved "GPT2InferenceModel object has no attribute 'generate'" error
by making GPT2InferenceModel classes explicitly inherit from GenerationMixin.

Changes:
Core model files patched:
- TTS/tts/layers/xtts/gpt_inference.py: Added GenerationMixin inheritance to XTTS GPT2 model
- TTS/tts/layers/tortoise/autoregressive.py: Added GenerationMixin inheritance to Tortoise GPT2 model

Documentation updated:
- TROUBLESHOOTING.md: Added transformers 4.50+ compatibility section
- STREAMLIT_QUICKSTART.md: Added transformers version warning
- STREAMLIT_TTS_GUIDE.md: Added transformers compatibility notes

Background:
Transformers v4.50+ changed the inheritance structure. PreTrainedModel no longer
automatically inherits from GenerationMixin. Models that use generation methods
like .generate() must now explicitly inherit from GenerationMixin.

Both GPT2InferenceModel classes now inherit from:
  - GPT2PreTrainedModel (base model functionality)
  - GenerationMixin (generation capabilities like .generate())

This maintains compatibility with transformers >=4.50 while preserving all
existing functionality.

claude committed:
Fix audio preview playback and add voice cloning quality tips

Fixed critical bug where audio preview wouldn't play in browser by
passing audio bytes instead of file path to st.audio().

Changes:
- streamlit_tts_app.py:
  * Fixed st.audio() to use audio bytes instead of temp file path
  * Read audio file once and reuse bytes for both player and download
  * Added expandable "Tips for Best Voice Cloning Quality" section
  * Tips include optimal audio length, quality, and format guidance

Background:
The st.audio() function treats string arguments as URLs. Passing a
temp file path like "/tmp/xyz.wav" created a broken reference that
the browser couldn't fetch, resulting in no audio playback or
corrupted output.

By reading the audio file as bytes and passing that to st.audio(),
the audio is properly embedded and playable in the browser.

This also improves user guidance by adding in-UI tips for getting
high-quality voice cloning results, including:
- Optimal audio length (6-10 seconds)
- Using multiple samples of same voice
- Audio quality requirements
- What to avoid (noise, music, multiple speakers)

claude committed:
Add voice cloning quality tips and clarify XTTS v2 requirements

Improved UI to clearly communicate that XTTS v2 requires reference audio
and added comprehensive quality tips for better voice cloning results.

Changes:
- Added prominent info box explaining XTTS v2 requires reference audio
- Updated file uploader label to show "*Required*"
- Expanded quality tips with more specific guidance
- Added warning when no reference audio is uploaded
- Disabled generate button when reference audio is missing for English
- Added helpful caption explaining why button is disabled
- Added suggestions for where to find good reference audio

Background:
XTTS v2 is a zero-shot voice cloning model that doesn't have default
speakers. It REQUIRES reference audio to function. The "whimpering and
moaning" audio quality issue was caused by either:
1. Poor quality reference audio (compressed, noisy, too short)
2. Attempting to generate without reference audio (error)

This update makes the requirements crystal clear and helps users
understand what constitutes good reference audio for optimal results.

eginhard commented:
Note that the various issues with dependencies are already fixed in [our fork](https://github.com/idiap/coqui-ai-TTS), available via `pip install coqui-tts`
